.github/workflows/check_critic.yml

```diff
@@ -0,0 +1,15 @@
+---
+name: perlcritic
+on: [push, pull_request]
+
+jobs:
+  perlcritic_checks:
+    runs-on: ubuntu-latest
+    container:
+      image: registry.opensuse.org/devel/openqa/containers/os-autoinst_dev
+    steps:
+      - uses: actions/checkout@v4
+      - name: Static analysis
+        run: |
+          git config --global --add safe.directory '*'
+          ./external/os-autoinst-common/tools/perlcritic --quiet $(git ls-files -- '*.p[ml]')
```

Comment:
As mentioned in the other PR, please use perlcritic like this:
https://github.com/os-autoinst/os-autoinst-common/blob/master/.github/workflows/perl-critic.yml
You don't need to pass it the list of files.